libraries/push/api/src/main/kotlin/io/element/android/libraries/push/api/battery/BatteryOptimizationState.kt

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright 2025 New Vector Ltd.
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only OR LicenseRef-Element-Commercial
+ * Please see LICENSE files in the repository root for full details.
+ */
+
+package io.element.android.libraries.push.api.battery
+
+import android.app.Activity
+
+data class BatteryOptimizationState(
+    val shouldDisplayBanner: Boolean,
+    val dismiss: () -> Unit,
+    val openSettings: (Activity?) -> Unit,
```

Comment:
Use EventSink instead please?
Also we don't want Activity reference in our `State`, the activity should be grabbed in the dependency if possible. By the way, appContext is not enough here?